AzureMonitorAgent/agent.py

```diff
@@ -47,6 +47,7 @@
 import xml.dom.minidom
```

Comment:
to be safe, I would recommend a e2e test in Canary where we first do install with no args, then call install again with proxy or any other settings to see if defaluts file is correct

Reply:
testing this now locally

Reply:
@simathih on my test machine, this works as expected;
1. -enable with null public settings in 0.settings
2. add 1.settings with proxy content
3. add random # comment in defaults file
4. -enable again
5. defaults file now has proxy content and # comment is gone